crates/ide/src/move_item.rs

```diff
@@ -64,16 +68,39 @@ fn find_ancestors(item: SyntaxElement, direction: Direction) -> Option<TextEdit>
         .chain(root.ancestors())
         .find(|ancestor| movable.contains(&ancestor.kind()))?;
 
-    move_in_direction(&ancestor, direction)
+    move_in_direction(&ancestor, direction, range)
 }
 
-fn move_in_direction(node: &SyntaxNode, direction: Direction) -> Option<TextEdit> {
-    let sibling = match direction {
-        Direction::Up => node.prev_sibling(),
-        Direction::Down => node.next_sibling(),
-    }?;
+fn move_in_direction(
+    node: &SyntaxNode,
+    direction: Direction,
+    range: TextRange,
+) -> Option<TextEdit> {
+    match_ast! {
+        match node {
+            ast::ArgList(it) => swap_sibling_in_list(it.args(), range, direction),
+            _ => Some(replace_nodes(node, &match direction {
+                Direction::Up => node.prev_sibling(),
+                Direction::Down => node.next_sibling(),
+            }?))
+        }
+    }
+}
 
-    Some(replace_nodes(node, &sibling))
+fn swap_sibling_in_list<'i, A: AstNode + Clone, I: Iterator<Item = A>>(
```

Comment:
Attempt of implementation with streaming iterator has failed, yet I forgot to remove lifetime